sunpy/visualization/animator/wcs.py

```diff
@@ -0,0 +1,241 @@
+    * 'right': next step on active slider.
+    * 'top': change the active slider up one.
+    * 'bottom': change the active slider down one.
+    * 'p': play/pause active slider.
+
+    Parameters
+    ----------
+    data: `numpy.ndarray`
+        The data to be visualized.
+    wcs: `astropy.wcs.wcsapi.BaseLowLevelWCS`
+        The world coordinate object associated with the array.
+    slices: `tuple` or `list`
+        A list specifying which axes of the array should be plotted on which
+        axes. The list should be the same length as the number of pixel
+        dimensions with ``'x'`` and (optionally) ``'y'`` in the elements
+        corresponding to the axes to be plotted. If only ``'x'`` is present a
+        line plot will be drawn. All other elements should be `0`.
+    coord_params: `dict`, optional
+        This dict allows you to override
+        `~astropy.visualization.wcsaxes.WCSAxes` parameters for each world
+        coordinate. The keys of this dictionary should be a value which can be
+        looked up in ``WCSAxes.coords`` (i.e. ``em.wl`` or ``hpln``) and the
+        values should be a dict which supports the following keys, and passes
+        their values to the associated `~astropy.visualization.wcsaxes.WCSAxes`
+        methods.
+
+        * ``format_unit``: `~astropy.visualization.wcsaxes.CoordinateHelper.set_format_unit`
+        * ``major_formatter``: `~astropy.visualization.wcsaxes.CoordinateHelper.set_major_formatter`
+        * ``axislabel``: `~astropy.visualization.wcsaxes.CoordinateHelper.set_axislabel`
+        * ``grid``: `~astropy.visualization.wcsaxes.CoordinateHelper.grid` (The value should be a dict of keyword arguments to ``grid()`` or `True`).
+    ylim: `tuple` or `None`, optional
+       The yaxis limits to use when drawing a line plot, if not specified
+       defaults to the data limits.
+    ylabel: `string`, optional
+       The yaxis label to use when drawing a line plot. Setting the label on
+       the y-axis on an image plot should be done via ``coord_params``.
+
+    """
+
+    def __init__(self, data, wcs, slices, coord_params=None, ylim=None, ylabel=None, **kwargs):
+        if not isinstance(wcs, BaseLowLevelWCS):
+            raise ValueError("A WCS object should be provided that implements the astropy WCS API.")
+        if wcs.pixel_n_dim is not data.ndim:
+            raise ValueError("Dimensionality of the data and WCS object do not match.")
+        if len(slices) != wcs.pixel_n_dim:
+            raise ValueError("slices should be the same length as the number of pixel dimensions.")
+        if "x" not in slices:
+            raise ValueError("slices should contain at least 'x' to indicate the axis to plot on the x axis.")
+
+        self.plot_dimensionality = 1
+
+        image_axes = [slices[::-1].index("x")]
+        if "y" in slices:
+            image_axes.append(slices[::-1].index("y"))
+            self.plot_dimensionality = 2
```

Comment:
Does the way this is done mean that you can't customize which dimensions are on the x and y axis for image animations?

Reply:
It does not. You just specify `('y', 'x')` rather than `('x', 'y')` to specify the order.